skbio/metadata/_testing.py

```diff
@@ -904,3 +905,245 @@ def test_has_positional_metadata(self):
+        self.assertEqual(self.im, obj.interval_metadata)
+
+        # Update existing column.
+        obj.interval_metadata._intervals[0].metadata['gene'] = 'sagB'
+        self.assertNotEqual(obj.interval_metadata, self.im)
+        self.im._intervals[0].metadata['gene'] = 'sagB'
+        self.assertEqual(obj.interval_metadata, self.im)
+
+        # Add new interval feature.
+        obj.interval_metadata.add(**self.intvls[1])
+        self.im.add(**self.intvls[1])
+        self.assertEqual(obj.interval_metadata, self.im)
+
+    def test_interval_metadata_setter_makes_copy(self):
+        intvl = self.im.add(**self.intvls[1])
+        obj = self._interval_metadata_constructor_(self.upper_bound)
+        obj.interval_metadata = self.im
+
+        self.assertEqual(obj.interval_metadata, self.im)
+        self.assertIsNot(obj.interval_metadata, self.im)
+
+        # Changing mutable value of metadata of the old interval
+        # also changes obj.
+        intvl.metadata['gene'].append('b')
+        self.assertEqual(obj.interval_metadata, self.im)
+
+        # Changing old interval doesn't change obj
+        intvl.locations = [(3, 6)]
+        self.assertNotEqual(obj.interval_metadata, self.im)
+
+    def test_interval_metadata_setter_invalid_type(self):
+        self.im.add(**self.intvls[0])
+        obj = self._interval_metadata_constructor_(self.upper_bound, self.im)
+
+        for i in [2, None, '', {}, []]:
+            with self.assertRaisesRegex(
+                    TypeError,
+                    'You must provide `IntervalMetadata` object'):
+                obj.interval_metadata = i
+
+        self.assertEqual(self.im, obj.interval_metadata)
+
+    def test_interval_metadata_deleter_empty(self):
+        obj = self._interval_metadata_constructor_(self.upper_bound, self.im)
+
+        del obj.interval_metadata
+        self.assertIsNone(obj._interval_metadata)
+        self.assertFalse(obj.has_interval_metadata())
+
+        # Delete again. test idempotent
+        del obj.interval_metadata
+        self.assertIsNone(obj._interval_metadata)
+        self.assertFalse(obj.has_interval_metadata())
+
+    def test_interval_metadata_deleter(self):
+        self.im.add(**self.intvls[0])
+        obj = self._interval_metadata_constructor_(self.upper_bound, self.im)
+
+        del obj.interval_metadata
+        self.assertIsNone(obj._interval_metadata)
+        self.assertFalse(obj.has_interval_metadata())
+
+    def test_has_interval_metadata(self):
+        obj = self._interval_metadata_constructor_(self.upper_bound, self.im)
+        self.assertFalse(obj.has_interval_metadata())
+
+        obj = self._interval_metadata_constructor_(self.upper_bound, self.im)
+        self.assertFalse(obj.has_interval_metadata())
+
+        self.im.add([(0, 1)])
+        obj = self._interval_metadata_constructor_(self.upper_bound, self.im)
+        self.assertTrue(obj.has_interval_metadata())
+
+    def test_reverse(self):
```

Comment:
Sorry for requesting this test, it's unnecessary b/c we're already asserting that the IntervalMetadata object is constructed correctly in several different situations, and that the object's axis length is respected in the IntervalMetadata object. We're not testing specific methods on IntervalMetadata here, and `reverse` isn't a special case. Feel free to remove this test.